### DISCUSSION TITLE
Any recommendations for bumping up security?

### DISCUSSION THREAD

**Discussion by Ascar699** [dani-garcia/vaultwarden#3191] [General]
Hello, everyone.

I am seeking suggestions and recommendations on whether anything should be done to increase VW resilience in view of the recent discovery of LastPass vulnerabilities and active discussion of server-side PBKD2 iterations.

[Bitwarden design flaw: Server side iterations](https://palant.info/2023/01/23/bitwarden-design-flaw-server-side-iterations/)
[Bitwarden password vaults targeted in Google ads phishing attack](https://www.bleepingcomputer.com/news/security/bitwarden-password-vaults-targeted-in-google-ads-phishing-attack/)

**Comment by BlackDex** [comment]
Just change your KDF iterations to 600_000 and be done with it.
Also, if you use the `testing` tagged Vaultwarden images it will have a default of 600_000 on the server side also. And it will migrate existing users from the previous 100_000 (server side) to 600_000 (server side) on login. The server-side KDF is only used for the master-password hash, which is sent by the clients to verify the user has typed the correct password or not. All other encryption related stuff is handled by the clients.

**Reply by Ascar699** [reply]
Thank you, @BlackDex.  I checked the environment template and found the following:

##Controls the PBBKDF password iterations to apply on the server
##The change only applies when the password is changed
#PASSWORD_ITERATIONS=100000


I did not have this string in my docker-compose.yaml file at all.  If I insert it into my yaml file now when the container is down and then restart - do I need to change my master password as the comment suggests?

**Reply by BlackDex** [reply]
If you use the `testing` tagged images it will update the user's password hashing to 600_000. With the current `latest` tagged images it will not change anything for current users, only for new users.

600_000 is the default in newer versions of Vaultwarden (not yet released as stable/latest).
